solr/modules/hadoop-auth/src/test/org/apache/solr/security/hadoop/TestZkAclsWithHadoopAuth.java

```diff
@@ -106,6 +107,10 @@ public void testZkAcls() throws Exception {
       String zkHost = cluster.getSolrClient().getClusterStateProvider().getQuorumHosts();
       String zkChroot = zkHost.contains("/") ? zkHost.substring(zkHost.indexOf("/")) : null;
       walkZkTree(keeper, zkChroot, "/");
+    } finally {
+      // NOTE: cannot use try-with-resources, because `ZooKeeper.close()` without shutdownTimeout
+      // does not join on (and can leak) connection threads.
```

Comment:
I'm not sure this logic belongs in `SolrZkClient`. We should use `try w/ resources` if possible. Can we override/extend `ZooKeeper` with a `SolrZooKeeper` or something that always closes Zookeeper the way we want? That way we get the benefit of try w/ resources and still get Zookeeper to close.